compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,51 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// Make sure that we are in the condition to suggest the blanket implementation.
+    fn maybe_lint_blanket_trait_impl<T: rustc_errors::EmissionGuarantee>(
+        &self,
+        self_ty: &hir::Ty<'_>,
+        diagnostic: &mut DiagnosticBuilder<'_, T>,
+    ) {
+        let tcx = self.tcx();
+        let parent_id = tcx.hir().get_parent_item(self_ty.hir_id);
+        if let hir::Node::Item(hir::Item {
+            kind:
+                hir::ItemKind::Impl(hir::Impl {
+                    self_ty: impl_self_typ, of_trait: Some(of_trait_ref), generics, ..
+                }),
+            ..
+        }) = tcx.hir().get_by_def_id(parent_id) && self_ty.hir_id == impl_self_typ.hir_id
+        {
+            let of_trait_span = of_trait_ref.path.span;
+            // check if the trait has generics, to make a correct suggestion
+            let of_target_span = if let Some(span) = generics.span_for_param_suggestion() {
+                span
+            } else {
+                of_trait_span
+            };
+
+            let of_typ_is_local = of_trait_ref.trait_def_id().map_or(false, |def_id| def_id.is_local());
+            if of_typ_is_local {
+                let of_trait_name = tcx.sess.source_map().span_to_snippet(of_trait_span).unwrap();
+                let impl_trait_name = tcx.sess.source_map().span_to_snippet(self_ty.span).unwrap();
+                let blanket_msg = format!(
+                    "use a blanket implementation to implement {} for all types that also implement {}",
```

Comment:
Surround the traits in backticks:

```suggestion
                    "use a blanket implementation to implement `{}` for all types that also implement `{}`",
```